File: exercises/beecrowd/1117/1117.ts
// https://judge.beecrowd.com/pt/problems/view/1117
import * as fs from "fs";

function main(): void {
  const inputs = fs.readFileSync('/dev/stdin', 'utf8').trim().split('\n').map(Number);

  let validCount = 0;
  let sum = 0;

  for (const grade of inputs) {
    if (grade < 0 || grade > 10) {
      console.log("nota invalida");
    } else {
      sum += grade;
      validCount++;
    }

    if (validCount === 2) {
      const average = sum / 2;
      console.log(`media = ${average.toFixed(2)}`);
      break;
    }
  }
}

main();

File: exercises/beecrowd/1117/1117.go
// https://judge.beecrowd.com/pt/problems/view/1117
package main

import (
	"fmt"
)

func main() {
	var grade float64
	validCount := 0
	var sum float64

	for validCount < 2 {
		fmt.Scan(&grade)
		if grade < 0.0 || grade > 10.0 {
			fmt.Println("nota invalida")
		} else {
			sum += grade
			validCount++
		}
	}

	average := sum / 2.0
	fmt.Printf("media = %.2f\n", average)
}
